src/core/abstractreceiver.h

```diff
@@ -0,0 +1,59 @@
+/***************************************************************************
+ abstractreceiver.h - AbstractReceiver
+
+ ---------------------
+ begin                : 22.05.2022
+ copyright            : (C) 2022 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef ABSTRACTRECEIVER_H
+#define ABSTRACTRECEIVER_H
+
+#include "gnsspositioninformation.h"
+
+#include <QAbstractSocket>
+#include <QObject>
+
+class AbstractReceiver : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY( GnssPositionInformation lastGnssPositionInformation READ lastGnssPositionInformation NOTIFY lastGnssPositionInformationChanged )
+    Q_PROPERTY( QAbstractSocket::SocketState socketState READ socketState NOTIFY socketStateChanged )
+    Q_PROPERTY( QString socketStateString READ socketStateString NOTIFY socketStateStringChanged )
+
+  public:
+    explicit AbstractReceiver( QObject *parent = nullptr )
+      : QObject( parent ) {}
+    virtual ~AbstractReceiver() = default;
+
+    virtual void disconnectDevice() = 0;
```

Comment:
Can we add virtual private methods and call them from these (non virtual) public ones?